### PR TITLE
Vote page: next iteration

### LINKED ISSUE #99: Vote page: Implement next iteration of proposal view
from element:

thinking through the voting page and drawing some things out. Looking at each proposal as its own unit, brainstorming ways to display any cross-chain complexity in the most simple way. I'm currently thinking through a 2 row approach, like we're already sorta using, except with L2 on top:

L2: [ container showing vote breakdown ]
L1: [ container showing vote breakdown ]

in the L2 container, I want to show what votes have been casted on L2 but not yet bridged. these should show as like a "pending" color, maybe like a lighter green or striped green for the for votes and likewise but red for the against. I also want to show which votes there have been bridged, so those could be a solid green/red or something.

actually though i'm thinking this l2 section can be a little smaller, almost like a "tooltip" size, because I want to show the same info in the L1 container.

for votes: we'll see 3 bars total, stacked next to each other: the amount of votes from regular L1 voters, the amount of bridged votes, and then in a lighter green the amount of "pending" (not yet bridged) for votes.
against votes: same thing
and yeah we would probably do the same for abstain just to show the complete picture.

these containers would also show stuff like vote power, and stuff such as the total amount of votes that have not been cast yet, but i wanted to focus just on getting the L2/L1 vote display right.

also thinking that we can use more vertical space (within reason) if we need to. there's no requirement that the proposals have to be as tightly stacked as say on tally. ofc we could opt for expand/collapse too, depending where we land at

### PR DESCRIPTION
**Pull request by wildmolasses**
Closes #99

### CODE REVIEW

**Comment by wildmolasses** [comment]
Current dependencies on/for this PR:
* main
  * **PR #118** <a href="https://app.graphite.dev/github/pr/ScopeLift/l2-flexible-voting-frontend/118?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #119** <a href="https://app.graphite.dev/github/pr/ScopeLift/l2-flexible-voting-frontend/119?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/ScopeLift/l2-flexible-voting-frontend/119?utm_source=stack-comment).

File: components/ProposalRow.tsx [hunk 1]
```diff
@@ -1,146 +1,337 @@
...
+      'text-green-500 text-green-300 text-red-500 text-red-300 text-gray-500 text-gray-300 bg-green-500 bg-green-300 bg-red-500 bg-red-300 bg-gray-500 bg-gray-300';
+    return (
+      <>
+        <div className={'flex w-full items-center text-' + color + '-500'}>
+          <span className="font-bold text-xs">
+            {nFormatter(+formatUnits(votes, l2.tokenDecimals))}
+          </span>
+          <span className={'ml-1 text-xs ' + 'text-' + color + '-300'}>
+            {bridged > BigInt(0)
+              ? `(${nFormatter(+formatUnits(bridged, l2.tokenDecimals))} bridged from L2)`
```
Comment (alexkeating) [inline components/ProposalRow.tsx:55]: Should the tokenDecimals be from L2 or from L1?

File: components/ProposalRow.tsx [hunk 2]
```diff
@@ -1,146 +1,337 @@
...
+  });
+
+  const l1AbstainVoteBar = makeL1VoteBar({
+    totalVotes,
+    votes: votes.l1.abstainVotes || BigInt(0),
+    bridged: votes.l2Bridged?.abstainVotes || BigInt(0),
+    color: 'gray',
   });
+
+  const makeL2VoteBar = ({
```
Comment (alexkeating) [inline components/ProposalRow.tsx:99]: It may be worth merging this with the L1 component in a future pr.

File: hooks/useProposals.ts [hunk 3]
```diff
@@ -77,8 +88,12 @@ const createFetcher =
       createdLogs.map(async (event) => {
         const args = event.args;
         const proposalId = args[0]?.toString() || '0';
+        const bridgedVotes = voteBridgedLogs?.find(
+          (event) => event.args.proposalId?.toString() === proposalId
+        )?.args;
+        console.log(voteBridgedLogs);
+        console.log(bridgedVotes);
```
Comment (alexkeating) [inline hooks/useProposals.ts:95]: Excess console logs

File: hooks/useProposals.ts [hunk 4]
```diff
@@ -191,8 +210,15 @@ export const useProposals = () => {
         description: proposal.description,
         isBridged: Boolean(l2Proposal),
         status: {
-          l1: proposal.isCancelled ? 'cancelled' : 'open',
-          l2: l2Proposal?.isCancelled ? 'cancelled' : 'open',
+          l1: proposal.isCancelled
+            ? 'cancelled'
+            : l1Block && l1Block > BigInt(proposal.endBlock)
+            ? 'closed'
+            : 'open',
```
Comment (alexkeating) [inline hooks/useProposals.ts:217]: Not a huge fan of nested ternaries. One of these could be abstracted to a variable

**Review by alexkeating — COMMENTED** [review]
Working well looks good some small comments

File: hooks/useProposals.ts (hunk 4 above)
Comment (wildmolasses) [reply hooks/useProposals.ts:217]: i'd be down. the proposal statuses need a total revamp, so i think i'll defer this change until then. tracking in #122

File: components/ProposalRow.tsx (hunk 1 above)
Comment (wildmolasses) [reply components/ProposalRow.tsx]: thanks for the catch

**Pull request merged**